nifi-extension-bundles/nifi-slack-bundle/nifi-slack-processors/src/test/java/org/apache/nifi/processors/slack/TestGetSlackReaction.java

```diff
@@ -0,0 +1,196 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.nifi.processors.slack;
+
+import com.slack.api.bolt.App;
+import com.slack.api.methods.MethodsClient;
+import com.slack.api.methods.request.reactions.ReactionsGetRequest;
+import com.slack.api.methods.response.reactions.ReactionsGetResponse;
+import com.slack.api.model.Reaction;
+
+import org.apache.nifi.util.MockFlowFile;
+import org.apache.nifi.util.TestRunner;
+import org.apache.nifi.util.TestRunners;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.time.Duration;
+import java.util.Arrays;
+import java.util.List;
+
+import static java.util.Collections.emptyList;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
+@ExtendWith(MockitoExtension.class)
+public class TestGetSlackReaction {
+    public static final String TEST_MESSAGE_TS = "123456.789";
+    public static final String TEST_CHANNEL_ID = "cid1";
+    private TestRunner testRunner;
+    GetSlackReaction processor;
+    @Mock
+    private MethodsClient clientMock;
+
+    @BeforeEach
+    public void setup() {
+        processor = new GetSlackReaction() {
+            @Override
+            protected MethodsClient initializeClient(final App slackApp) {
+                return clientMock;
+            }
+        };
+
+        testRunner = TestRunners.newTestRunner(processor);
+        testRunner.setProperty(GetSlackReaction.ACCESS_TOKEN, "token");
+        testRunner.setProperty(GetSlackReaction.CHANNEL_ID, TEST_CHANNEL_ID);
```

Comment:
This test is helpful to better understand what the proposed use case is.  Here it shows the user supplying a specific processor property value for the channel and timestamp of the message to gather reactions for.  This is not a valid use case for NiFi.  The valid use case would be for these attributes to come from the incoming flow file.